### PR TITLE
gzip compression for data (post and responses)

### PR DESCRIPTION
**Pull request by KEClaytor**
Hello, I have enabled gzip content encoding in this branch. I have tested it against my own influx application and it seems to work. However, I am unsure how to properly do the tests.

I have included some tests, however, they do not actually track integration with the influxdb application. For instance, the tests pass when I use zlib do do the compression, even though that does not work in my application. Can you advise where / how to do proper implementation tests?

If you would like to incorporate it into the main branch, I would certainly find it useful and others may as well (see: #718).

### CODE REVIEW

**Comment by KEClaytor** [comment]
Okay, figured out zlib so that it's compatible with python 2.7.
Local tests seem to pass, but I'm not testing against a range of influxdb versions (only 1.7).
I basically duplicated all server tests but enabled gzip=True.
There are no tests for influxdb08 - are they needed?

File: influxdb/client.py [hunk 1]
```diff
@@ -66,8 +67,9 @@ class InfluxDBClient(object):
         as a single file containing the private key and the certificate, or as
         a tuple of both files’ paths, defaults to None
     :type cert: str
-
     :raises ValueError: if cert is provided but ssl is disabled (set to False)
+    :parm gzip: use gzip content encoding to compress requests
```
Comment (xginn8) [inline influxdb/client.py]: this should be `param`, and should come before the `:raises` block for consistency

Comment (KEClaytor) [reply influxdb/client.py]: No problem, thanks!

**Comment by KEClaytor** [comment]
Interestingly I noticed that the zlib compressed data wasn't actually being added to InfluxDB. Switching back to gzip and manually ensuring that data is posted.

**Comment by f0rtung** [comment]
Is there anyone alive?

**Comment by cgiraldo** [comment]
Any chance to see this PR merged?

**Pull request merged**